src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -0,0 +1,368 @@
+        mockSpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
+          cb(null, {
+            userName: 'S1234567A',
+          }),
+        )
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
+          .query({ captchaResponse: 'null' })
+          .set('Cookie', ['jwtSp=mockJwt'])
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual({
+          message: 'Form submission successful.',
+          submissionId: expect.any(String),
+        })
+      })
+
+      it('should return 401 when submission does not have JWT', async () => {
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
+          .query({ captchaResponse: 'null' })
+
+        expect(response.status).toBe(401)
+        expect(response.body).toEqual({
+          message:
+            'Something went wrong with your login. Please try logging in and submitting again.',
+          spcpSubmissionFailure: true,
+        })
+      })
+
+      it('should return 401 when submission has the wrong JWT type', async () => {
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
+          .query({ captchaResponse: 'null' })
+          .set('Cookie', ['jwtCp=mockJwt'])
+
+        expect(response.status).toBe(401)
+        expect(response.body).toEqual({
+          message:
+            'Something went wrong with your login. Please try logging in and submitting again.',
+          spcpSubmissionFailure: true,
+        })
+      })
+
+      it('should return 401 when submission has invalid JWT', async () => {
+        mockSpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
```

Comment:
nit: add comment here to draw attention?

Reply:
That's also why i do the Arrange-Act-Assert comments, help group the tasks, but that's personal preference i guess